win_installer/_base.sh

```diff
@@ -24,183 +24,99 @@ SDK_PLATFORM="win32"
-    export WINEDEBUG=-all
-    export WINEDLLOVERRIDES="mscoree,mshtml="
-
-    # try to limit the effect on the host system when installing with wine.
-    export HOME="$BUILD_ENV"/home
-    export XDG_DATA_HOME="$HOME"/.local/share
-    export XDG_CONFIG_HOME="$HOME"/.config
-    export XDG_CACHE_HOME="$HOME"/.cache
-    
-    # crashes on wine 1.8
-    #export DISPLAY_SAVED=$DISPLAY
-    #export DISPLAY=be_quiet_damnit
-
-    mkdir -p "$WINEPREFIX"
-    wine wineboot -u
+function build_python {
+  "${BUILD_ROOT}"/"${MINGW}"/bin/"${PYTHON_ID}".exe "$@"
 }
 
-function init_build_env {
-    # create a fresh build env and link the binaries in
-    rm -Rf "$BUILD_ENV"
-    mkdir "$BUILD_ENV"
-    ln -s "$SDK_BIN" "$BUILD_ENV"/bin
+function build_makensis {
+  "${BUILD_ROOT}"/"${MINGW}"/bin/makensis.exe "$@"
 }
 
+function create_root {
+  mkdir -p "${BUILD_ROOT}"
 
-function extract_deps {
-  [ -f "$SDK_DATA"/_extract_deps.sh ] && source "$SDK_DATA"/_extract_deps.sh
-  [ -f "$TARGET_DATA"/_extract_deps.sh ] && source "$TARGET_DATA"/_extract_deps.sh
-}
+  mkdir -p "${BUILD_ROOT}"/var/lib/pacman
+  mkdir -p "${BUILD_ROOT}"/var/log
+  mkdir -p "${BUILD_ROOT}"/tmp
 
-function setup_deps {
-  [ -f "$SDK_DATA"/_setup_deps.sh ] && source "$SDK_DATA"/_setup_deps.sh
-  [ -f "$TARGET_DATA"/_setup_deps.sh ] && source "$TARGET_DATA"/_setup_deps.sh
-  
-  install_pydeps
+  build_pacman --noconfirm -Syu
+  build_pacman --noconfirm -S base
 }
 
-function _install_pydep {
-  # arg1: requirements.txt, arg2: output dir
-  local PYTHON="$PYDIR"/python.exe
-  if [ -f "$1" ]; then
-    sed 's/^#bindep\s*//' "$1" > "$1".tmp
-    (wine "$PYTHON" -m pip install -f "$2" -r "$1".tmp)
-    rm "$1".tmp
+function install_deps {
+  if [ -n "$1" ]; then
+    build_pacman --noconfirm -S $1
   fi
+    
 }
 
 function install_pydeps {
-  local PYTHON="$PYDIR"/python.exe
-  _install_pydep "$SDK_DATA"/requirements.txt "$SDK_BIN"
-  _install_pydep "$TARGET_DATA"/requirements.txt "$TARGET_BIN"
-}
-
-
-function setup_sdk {
-    SDK="$BUILD_ENV"/python-gtk3-gst-sdk
-    mkdir "$SDK"
-
-    # launchers, README
-    ln -s "$SDK_SCRIPTS"/* "$SDK"
-    [ -d "$TARGET"/sdk_scripts ] && ln -s "$TARGET"/sdk_scripts/* "$SDK"
-    
-    # Create a clone script
-    echo "git clone $GIT_CLONE_URL" > "$SDK"/clone.bat
-
-    # bin deps
-    ln -s "$DEPS" "$SDK"/deps
-    ln -s "$PYDIR" "$SDK"/python
-    ln -s "$GITDIR" "$SDK"/git
-
-    # link to base dir
-    ln -s "$SDK" "$TARGET"/_sdk
-
-    # create the distributable archive
-    echo "Creating distribution tarball (ctrl-c if you don't care)"
-    tar --dereference -zcvf "$TARGET"/python-gtk3-gst-win32-sdk.tar.gz _sdk/ \
-        --exclude=_sdk/_wine_prefix &> /dev/null
+  # Since we need pywin32_ctypes to install *before* PyInstaller for Exaile,
+  # we will iterate over the requirements file and install one by one, which
+  # guarantees the order
+  while IFS='' read -r line || [[ -n "$line" ]]; do
+    build_pip install $line
```

Comment:
Yes that indeed is the issue, I pinned `mutagen` to a certain version, and because there already existed one for the user, `pip` uninstalls the existing version first.

This is a tough one to fix, since I can't use `virtualenv` (it doesn't work in msys2), and using flags like `--root` or `--ignore-installed` breaks Pyinstaller discovery of pywin32ctypes